peps/pep-0796.rst

```diff
@@ -0,0 +1,322 @@
+virtual environment once, cache it, and then re-use it as-is to CI jobs.
+
+
+Rationale
+=========
+
+Defining semantics for a relative ``home`` path is the chosen design for the
+following reasons.
+
+First, it is a small change to interpreter startup, in particular of an
+unreliable behavior that isn't specified. Currently, relative paths are
+resolved to the process's current working directory, which makes them
+unreliable for use in practice.
+
+Second, for portable virtual environments, relative paths allow more
+efficient, simple, and correct reproduction of virtual environments between
+hosts. This is because they can be copied as-is to different locations. Some
+example capabilities this allows are:
+
+* A build farm creating (and caching) a virtual environment, which is then
+  served to developers (e.g. Bazel).
+* Composing virtual environments together (e.g. venvstacks).
+* Installing multiple arbitrary virtual environments into a container to
+  save disk space.
+* Layering a virtual environment atop a container image for faster image
+  building.
+* Not needing virtual environment creation tools on the host that uses a
+  virtual environment.
+* Exact reproduction of an application's virtual environment between a
+  developer's host and a production host.
+
+Third, requiring an absolute path is inherently overly proscriptive. The
+interpreter itself doesn't care whether paths are relative or absolute, merely
+that they point to valid locations, so users should be given the ability to use
+a path of their choosing. Given a known anchor point, it's easy to transform a
+relative path to an absolute path and still retain predictable and reliable
+behavior that produces correct values.
+
+Fullying designing portable virtual environments
+------------------------------------------------
+
+This PEP purposely only focuses on the interpreter startup behavior to limit
+its scope. There are multiple implementations and many design questions for how
+to implement portable virtual environments work (e.g. what installers should
+do), but they are separate from the Python runtime initialization phase.
+
+
+Specification
+=============
+
+The ``home`` value in ``pyvenv.cfg`` is permitted to use a relative path value.
+These may contain parent-directory references outside of the virtual environment root
+directory.
+For example:
+
+* ``subdir/whatever/bin`` (a directory within the virtual environment)
+* ``./subdir/whatever/bin`` (same as above)
+* ``../../../../../elsewhere/runtime/bin`` (a directory outside the virtual
+  environment)
+
+Relative paths are relative to the directory containing :file:`pyvenv.cfg`.
+During interpreter startup (i.e. :file:`getpath.py`), the relative path is joined to the
+directory containing ``pyvenv.cfg`` to form an absolute path.
+Parent-directory references (``../``) and current
+directory references (``./``) are resolved syntactically (i.e. not resolving
```

Comment:
```suggestion
directory references (``./``) are resolved lexically (i.e. not resolving
```

(I don't mind if this isn't accepted, I've just always referred to this style of resolution as lexical rather than as syntactic: https://github.com/python/cpython/issues/124825)q